signature/mechanism_gpgme.go

```diff
@@ -133,12 +137,56 @@ func (m *gpgmeSigningMechanism) Sign(input []byte, keyIdentity string) ([]byte,
 	if err != nil {
 		return nil, err
 	}
+
+	passphraseFile := os.Getenv(PASSPHRASE_FILE_ENVIRONMENT)
+	if options != nil && options.PassphraseFile != "" {
+		passphraseFile = options.PassphraseFile
+	}
+	if passphraseFile != "" {
+		logrus.Debugf("Reading user-specified passphrase for gpgme signing from %s", passphraseFile)
+		ppf, err := os.Open(passphraseFile)
+		if err != nil {
+			return nil, err
+		}
+		defer ppf.Close()
+
+		// Read the *first* line in the passphrase file, just as gpg(1) does.
+		buf, err := bufio.NewReader(ppf).ReadBytes('\n')
+		if err != nil && !errors.Is(err, io.EOF) {
+			return nil, fmt.Errorf("reading passphrase file: %w", err)
+		}
```

Comment:
This is not GPGME-specific, so it should probably be in the shared `mechanism.go` (and ideally have tests).

Or move it out of `signature` to `pkg/cli` altogether, and then I don’t care about the tests quite as much :)